samples/apps/smallbank/app/smallbank.cpp

```diff
@@ -106,235 +169,338 @@ namespace ccfapp
           auto account_r = account_view->get(name);
           if (account_r.has_value())
           {
-            return make_error(
+            set_error_status(
+              args,
               HTTP_STATUS_BAD_REQUEST,
-              "Account already exists in accounts table: " + name);
+              fmt::format(
+                "Account already exists in accounts table: '{}'", name));
+            return;
           }
           account_view->put(name, acc_id);
 
           auto savings_r = savings_view->get(acc_id);
           if (savings_r.has_value())
           {
-            return make_error(
+            set_error_status(
+              args,
               HTTP_STATUS_BAD_REQUEST,
-              "Account already exists in savings table: " + name);
+              fmt::format(
+                "Account already exists in savings table: '{}'", name));
+            return;
           }
           savings_view->put(acc_id, savings_amt);
 
           auto checking_r = checking_view->get(acc_id);
           if (checking_r.has_value())
           {
-            return make_error(
+            set_error_status(
+              args,
               HTTP_STATUS_BAD_REQUEST,
-              "Account already exists in checkings table: " + name);
+              fmt::format(
+                "Account already exists in checkings table: '{}'", name));
+            return;
           }
           checking_view->put(acc_id, checking_amt);
         }
 
-        return make_success(true);
+        set_ok_status(args);
       };
 
-      auto balance = [this](Store::Tx& tx, nlohmann::json&& params) {
+      auto balance = [this](RequestArgs& args) {
         // Check the combined balance of an account
-        std::string name = params["name"];
-        auto account_view = tx.get_view(tables.accounts);
+        if (headers_unmatched(args))
+        {
+          set_unmatched_header_status(args);
+          return;
+        }
+
+        BankDeserializer bd(args.rpc_ctx->get_request_body().data());
+        auto name = to_string(bd.name());
```

Comment:
Can’t bd.name() be a string? Having to copy values out of the FB rather defeats the point of using it.

Edit: the names are integers, why is the table keyed by string? We should align.

Reply:
I don't know I was also wondering that. I can change the table to be keyed by int

Reply:
Can we check this with @ashamis? I think the strings were used to match the original SmallBank spec, where (IIRC) we lookup from `name:string` to `id:int`? Still, if that's true, then the account names we pass in everywhere should be `string` not `int`

Reply:
Yes Eddy is right. Smallbank schema is defined as 
Account(Name, CustomerID),
Saving(CustomerID, Balance), Checking(CustomerID, Balance). 

so name should be a string

Reply:
Ok I'll change the flatbuffer shcema to take a string for the account names instead of using integers

Reply:
Done